app/src/main/java/ar/com/wolox/android/cookbook/googlelogin/helper/GoogleHelper.kt

```diff
@@ -21,84 +23,92 @@ class GoogleHelper @Inject constructor(context: Context) {
         GoogleAccount(it)
     }
 
-    companion object {
-
-        /**
-         * Set google login action to the view
-         *
-         * @param fragment
-         *  the fragment where the button is used
-         *
-         * @param resultCode
-         *  the code to catch the activity result
-         */
-        fun setGoogleLoginAction(view: View, fragment: Fragment, resultCode: Int) {
-            // On click button, open Google activity
-            view.setOnClickListener {
-                fragment.startActivityForResult(GoogleHelper.getClient(fragment).signInIntent, resultCode)
-            }
+    /**
+     * Set google login action to the view.
+     *
+     * @param fragment
+     *  the fragment where the button is used.
+     *
+     * @param resultCode
+     *  the code to catch the activity result.
+     */
+    fun setGoogleLoginAction(view: View, fragment: Fragment, resultCode: Int) {
+        // On click button, open Google activity
+        view.setOnClickListener {
+            fragment.startActivityForResult(getClient(fragment).signInIntent, resultCode)
         }
+    }
 
-        /**
-         * Set google logout action to the view.
-         *
-         * @param fragment
-         *  the fragment where the button is used.
-         *
-         * @param onComplete
-         *  callback to be called on logout complete.
-         */
-        fun setGoogleLogoutAction(view: View, fragment: Fragment, onComplete: () -> Unit) {
-            view.setOnClickListener {
-                GoogleHelper.getClient(fragment).signOut().addOnCompleteListener(fragment.requireActivity()) {
-                    onComplete()
-                }
+    /**
+     * Set google logout action to the view.
+     *
+     * @param fragment
+     *  the fragment where the button is used.
+     *
+     * @param onComplete
+     *  callback to be called on logout complete.
+     */
+    fun setGoogleLogoutAction(view: View, fragment: Fragment, onComplete: () -> Unit) {
+        view.setOnClickListener { _ ->
+            getClient(fragment).signOut().addOnCompleteListener(fragment.requireActivity()) { _ ->
+                onComplete()
             }
         }
+    }
+
+    /**
+     * Get the google sign in client object.
+     *
+     * @param fragment
+     *  the fragment where the login/logout is.
+     */
+    private fun getClient(fragment: Fragment): GoogleSignInClient {
+        // Configure Google Sign-in and the GoogleSignInClient object
+        val gso =
+                GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                        .requestEmail()
+                        .build()
+        return GoogleSignIn.getClient(fragment.requireActivity(), gso)
+    }
 
+    /**
+     * Get error message from a code.
+     *
+     * @param errorCode
+     *  the error code.
+     */
+    fun getErrorMessage(context: Context, errorCode: Int?): String =
```

Comment:
The `GoogleHelper` should not be concerned with which `String` the `View` should display. That's a detail only the `View` should know.